### ISSUE TITLE
False positive on WordPress Plugin WP Mail SMTP

### ISSUE THREAD

**Issue by fabio-blanco** [coreruleset/wordpress-rule-exclusions-plugin#45]
### Description

I have installed a Nginx WAF with Modsecurity CRS. This WAF protects a backend WordPress.

One request from one of the plugins (`wp-mail-smtp`) stimulated a false positive on the Modsecurity with the rule id 933120. 

### How to reproduce the misbehavior

  1 - login on the WordPress Admin console, install and configure WP Mail SMTP. (I configured it with a Personal Gmail account)
  2 - Access the WP Mail SMTP configuration page > tools > Send test email
  3 - See the 403 message on the browser and find it on the logs

The same error can also be obtained on the page WP Mail SMTP configuration page > Settings > Miscellaneous
Just change anything and try to save it. 

### Logs

This is the meaningful part of the audit log:
```
{"message":"PHP Injection Attack: Configuration Directive Found","details":{"match":"Matched \"Operator `Pm' with parameter `=' against variable `MATCHED_VARS:ARGS:_wp_http_referer' (Value: `/wp-admin/admin.php?page=wp-mail-smtp' )","reference":"---stripped here---","ruleId":"933120","file":"/etc/modsecurity.d/owasp-crs-wordpress/rules/REQUEST-933-APPLICATION-ATTACK-PHP.conf","lineNumber":"112","data":"Matched Data: smtp found within MATCHED_VARS:ARGS:_wp_http_referer: /wp-admin/admin.php?page=wp-mail-smtp","severity":"2","ver":"OWASP_CRS/4.2.0","rev":"","tags":["application-multi","language-php","platform-multi","attack-injection-php","paranoia-level/1","OWASP_CRS","capec/1000/152/242"],"maturity":"0","accuracy":"0"}},{"message":"Inbound Anomaly Score Exceeded (Total Score: 5)","details":{"match":"Matched \"Operator `Ge' with parameter `5' against variable `TX:BLOCKING_INBOUND_ANOMALY_SCORE' (Value: `5' )","reference":"","ruleId":"949110","file":"/etc/modsecurity.d/owasp-crs-wordpress/rules/REQUEST-949-BLOCKING-EVALUATION.conf","lineNumber":"222","data":"","severity":"0","ver":"OWASP_CRS/4.2.0","rev":"","tags":["anomaly-evaluation","OWASP_CRS"],"maturity":"0","accuracy":"0"}}
```

And here from Nginx error log
```
2024/06/19 04:36:06 [error] 1071#1071: *553 [client x.x.x.x] ModSecurity: Access denied with code 403 (phase 2). Matched "Operator `Ge' with parameter `5' against variable `TX:BLOCKING_INBOUND_ANOMALY_SCORE' (Value: `5' ) [file "/etc/modsecurity.d/owasp-crs-wordpress/rules/REQUEST-949-BLOCKING-EVALUATION.conf"] [line "222"] [id "949110"] [rev ""] [msg "Inbound Anomaly Score Exceeded (Total Score: 5)"] [data ""] [severity "0"] [ver "OWASP_CRS/4.2.0"] [maturity "0"] [accuracy "0"] [tag "anomaly-evaluation"] [tag "OWASP_CRS"] [hostname "x.x.x.x"] [uri "/wp-admin/admin.php"] [unique_id "171878256689.980974"] [ref ""], client: x.x.x.x, server: mydomain.com, request: "POST /wp-admin/admin.php?page=wp-mail-smtp-tools&tab=test HTTP/2.0", host: "mydomain.com", referrer: "https://mydomain.com/wp-admin/admin.php?page=wp-mail-smtp-tools"
```

### Your Environment

* CRS version (e.g., v3.3.4): 4.2.0
* Paranoia level setting (e.g. PL1) : PL1
* ModSecurity version (e.g., 2.9.6): v3.0.12
* Web Server and version or cloud provider / CDN (e.g., Apache httpd 2.4.54): Nginx 1.26
* Operating System and version: Docker image owasp/modsecurity-crs:4.2.0-nginx-202405220605 running on a Ubuntu 22.04

### Confirmation

[x ] I have removed any personal data (email addresses, IP addresses,
    passwords, domain names) from any logs posted.

### Possible solution

I fixed the problem with the following exclusion:

```
SecRule REQUEST_URI "@beginsWith /wp-admin/admin.php?page=wp-mail-smtp" \
    "id:1001,\
    phase:1,\
    pass,\
    nolog,\
    ctl:ruleRemoveTargetById=933120;ARGS:_wp_http_referer"
```

**Comment by azurit** [comment]
@fabio-blanco Hi and thanks for reporting this. Unfortunately, WordPress exclusions does not support any WP plugins or themes and there are no plans for adding this. Problems such this must be resolved using custom exclusion rules.
